programs/validator-bonds/ON_CHAIN_ANALYSIS.md

```diff
@@ -73,29 +89,81 @@ with
+          WHEN BLOCK_ID >= 285120000 and BLOCK_ID <= 285551999 THEN 660
+          WHEN BLOCK_ID >= 285552000 and BLOCK_ID <= 285983999 THEN 661
+          WHEN BLOCK_ID >= 285984000 and BLOCK_ID <= 286415999 THEN 662
+          WHEN BLOCK_ID >= 286416000 and BLOCK_ID <= 286847999 THEN 663
+          WHEN BLOCK_ID >= 286848000 and BLOCK_ID <= 287279999 THEN 664
+          WHEN BLOCK_ID >= 287280000 and BLOCK_ID <= 287711999 THEN 665
+          WHEN BLOCK_ID >= 287712000 and BLOCK_ID <= 288143999 THEN 666
+          WHEN BLOCK_ID >= 288144000 and BLOCK_ID <= 288575999 THEN 667
+          WHEN BLOCK_ID >= 288576000 and BLOCK_ID <= 289007999 THEN 668
+          WHEN BLOCK_ID >= 289008000 and BLOCK_ID <= 289439999 THEN 669
+          WHEN BLOCK_ID >= 289440000 and BLOCK_ID <= 289871999 THEN 670
+          WHEN BLOCK_ID >= 289872000 and BLOCK_ID <= 290303999 THEN 671
+          WHEN BLOCK_ID >= 290304000 and BLOCK_ID <= 290735999 THEN 672
+          WHEN BLOCK_ID >= 290736000 and BLOCK_ID <= 291167999 THEN 673
+          WHEN BLOCK_ID >= 291168000 and BLOCK_ID <= 291599999 THEN 674
+          WHEN BLOCK_ID >= 291600000 and BLOCK_ID <= 292031999 THEN 675
+          WHEN BLOCK_ID >= 292032000 and BLOCK_ID <= 292463999 THEN 676
+          WHEN BLOCK_ID >= 292464000 and BLOCK_ID <= 292895999 THEN 677
+          WHEN BLOCK_ID >= 292896000 and BLOCK_ID <= 293327999 THEN 678
+          WHEN BLOCK_ID >= 293328000 and BLOCK_ID <= 293759999 THEN 679
+          WHEN BLOCK_ID >= 293760000 and BLOCK_ID <= 294191999 THEN 680
+          WHEN BLOCK_ID >= 294192000 and BLOCK_ID <= 294623999 THEN 681
+          WHEN BLOCK_ID >= 294624000 and BLOCK_ID <= 295055999 THEN 682
+          WHEN BLOCK_ID >= 295056000 and BLOCK_ID <= 295487999 THEN 683
+          WHEN BLOCK_ID >= 295488000 and BLOCK_ID <= 295919999 THEN 684
+          WHEN BLOCK_ID >= 295920000 and BLOCK_ID <= 296351999 THEN 685
+          WHEN BLOCK_ID >= 296352000 and BLOCK_ID <= 296783999 THEN 686
+          WHEN BLOCK_ID >= 296784000 and BLOCK_ID <= 297215999 THEN 687
+          WHEN BLOCK_ID >= 297216000 and BLOCK_ID <= 297647999 THEN 688
+          WHEN BLOCK_ID >= 297648000 and BLOCK_ID <= 298079999 THEN 689
+          WHEN BLOCK_ID >= 298080000 and BLOCK_ID <= 298511999 THEN 690
+          WHEN BLOCK_ID >= 298512000 and BLOCK_ID <= 298943999 THEN 691
+          WHEN BLOCK_ID >= 298944000 and BLOCK_ID <= 299375999 THEN 692
+          WHEN BLOCK_ID >= 299376000 and BLOCK_ID <= 299807999 THEN 693
+          WHEN BLOCK_ID >= 299808000 and BLOCK_ID <= 300239999 THEN 694
+          WHEN BLOCK_ID >= 300240000 and BLOCK_ID <= 300671999 THEN 695
+          WHEN BLOCK_ID >= 300672000 and BLOCK_ID <= 301103999 THEN 696
+          WHEN BLOCK_ID >= 301104000 and BLOCK_ID <= 301535999 THEN 697
+          WHEN BLOCK_ID >= 301536000 and BLOCK_ID <= 301967999 THEN 698
+          WHEN BLOCK_ID >= 301968000 and BLOCK_ID <= 302399999 THEN 699
+          ELSE 0 END AS epoch,
```

Comment:
erm.... wat? how about `floor(block_id / 432000)`

Reply:
Ok. From the time being I was convinced the epochs at the start of the Solana time being was of a different length and the simple division does not work. Being a wrong assumption it seems :sweat_smile:

Reply:
but even then, if you already know that epoch 650 starts at some specific slot, you could just do `floor((block_id - 280800000) / 432000) - 650`